packages/charts/src/Charts/SeverityLine/SeverityLine.js

```diff
@@ -0,0 +1,77 @@
+import './severity-line.scss';
+
+import React, { Component } from 'react';
+
+import PropTypes from 'prop-types';
+
+class SeverityLine extends Component {
+
+    constructor(props) {
+        super(props);
+        this._getSeverityValue = this._getSeverityValue.bind(this);
+        this.state = {
+            severity: 0
+        };
+    }
+
+    _getSeverityValue(value) {
+        const { config: { width } } = this.props;
+        const CHART_MAP = { 1: '1', 2: (width / 3), 3: (width * (2 / 3)), 4: (width - 1) };
+        this.setState({
+            severity: CHART_MAP[value]
```

Comment:
You can set the severity with proposed function as `chartMapper(width)?.[value]`

Reply:
Hey @karelhala! Thanks for the review 😁. Would 
```jsx 
useEffect(() => {
        value && setSeverity(CHART_MAP[value]);
    }, [value, setSeverity]);
```
be an acceptable way to format this change? I'm not 100% sure I understand how `chartMapper(width)?.[value]` works.

Reply:
@LI0131 it is effectively the same thing. the `?.` notations exist to avoid trying to access object property on an undefined variable and causing runtime crash.
```jsx
x = undefined
console.log(x?.foo) // will not fail but logs undefined
console.log(x.foo) // will thrown an error (trying to call property foo on undefined)

```

Reply:
@Hyperkid123 Oh awesome, thanks so much for clearing that up for me. That seems like it's super helpful 😁